src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -411,11 +410,31 @@ private boolean hasFallThroughComment(DetailAST currentCase) {
      * @return true if relief comment found
      */
     private boolean hasReliefComment(DetailAST ast) {
-        return Optional.ofNullable(getNextNonCommentAst(ast))
-                .map(DetailAST::getPreviousSibling)
-                .map(previous -> previous.getFirstChild().getText())
-                .map(text -> reliefPattern.matcher(text).find())
-                .orElse(Boolean.FALSE);
+        final DetailAST result = getNextNonCommentAst(ast);
+        boolean hasReliefComment = false;
+        if (result != null) {
+            DetailAST previous = result.getPreviousSibling();
+            boolean firstChildStatus = true;
+            final int line = previous.getLineNo();
+            int prevLineNumber = line;
+            while (previous != null && line == prevLineNumber && firstChildStatus) {
```

Comment:
`line == prevLineNumber` why does this condition matter?

Reply:
Also, what types of DetailASTs are we checking in here? Only comments? Can we reuse `getNextNonCommentAst` within this loop?